code/datums/components/blind_sense.dm

```diff
@@ -0,0 +1,89 @@
+/datum/component/blind_sense
+	///The range we can hear-ping things from
+	var/hear_range = 8
+	///List of things we can't sense
+	var/list/sense_blacklist
+	///The amount of time you can sense things for
+	var/sense_time = 5 SECONDS
+	///Reference to the users ears
+	var/obj/item/organ/ears/ears
+	///Type casted to access client, owner
+	var/mob/owner
+	///What texture we use
+	var/masked_texture = "blind_texture_blank"
+	///Do we bloom?
+	var/bloom = TRUE
+
+/datum/component/blind_sense/New(list/raw_args)
+	. = ..()
+	//Register signal for sensing voices
+	RegisterSignal(SSdcs, COMSIG_GLOB_LIVING_SAY_SPECIAL, PROC_REF(handle_hear))
+	//Register signal for sensing sounds
+	RegisterSignal(SSdcs, COMSIG_GLOB_SOUND_PLAYED, PROC_REF(handle_hear))
+	//typecast to access client
+	owner = parent
```

Comment:
Could this cause a harddel?

Reply:
I doubt the subsystem would get deleted. Right?

Reply:
I guess.